miso-web/src/main/webapp/scripts/form_sampleclass.js

```diff
@@ -0,0 +1,145 @@
+FormTarget = FormTarget || {};
+FormTarget.sampleclass = (function($) {
+
+  /*
+   * Expected config {
+   *   isAdmin: boolean
+   *   pageMode: string 'create' or 'edit'
+   * }
+   */
+
+  return {
+    getSaveUrl: function(sampleClass) {
+      return sampleClass.id ? Urls.rest.sampleClasses.update(sampleClass.id) : Urls.rest.sampleClasses.create;
+    },
+    getSaveMethod: function(sampleClass) {
+      return sampleClass.id ? 'PUT' : 'POST';
+    },
+    getEditUrl: function(sampleClass) {
+      return Urls.ui.sampleClasses.edit(sampleClass.id);
+    },
+    getSections: function(config, object) {
+      return [{
+        title: 'Sample Class Information',
+        fields: config.isAdmin ? getEditableFields(config, object) : getReadOnlyFields(config, object)
+      }];
+    },
+    confirmSave: function(object, saveCallback) {
+      object.parentRelationships = SampleClass.getParents();
+      object.childRelationships = SampleClass.getChildren();
+      saveCallback();
+    }
+  }
+
+  function getReadOnlyFields(config, object) {
+    $('#save').remove();
+    return [{
+      title: 'Sample Class ID',
+      data: 'id',
+      type: 'read-only',
+      getDisplayValue: function(sampleClass) {
+        return sampleClass.id || 'Unsaved';
+      }
+    }, {
+      title: 'Alias',
+      data: 'alias',
+      type: 'read-only'
+    }, {
+      title: 'Category',
+      data: 'sampleCategory',
+      type: 'read-only'
+    }, {
+      title: 'Subcategory',
+      data: 'sampleSubcategory',
+      getDisplayValue: function(sampleClass) {
+        return sampleClass.sampleSubcategory || 'n/a';
+      },
+      type: 'read-only'
+    }, {
+      title: 'Suffix',
+      data: 'suffix',
+      getDisplayValue: function(sampleClass) {
+        return sampleClass.suffix || 'n/a';
+      },
+      type: 'read-only'
+    }, {
+      title: 'Receipt Allowed?',
+      data: 'directCreationAllowed',
+      type: 'checkbox',
+      disabled: true
+    }, {
+      title: 'DNase Treatable?',
+      data: 'dnaseTreatable',
+      type: 'checkbox',
+      disabled: true
+    }, {
+      title: 'Archived',
+      data: 'archived',
+      type: 'checkbox',
+      disabled: true
+    }];
+  }
+
+  function getEditableFields(config, object) {
+    return [
+        {
+          title: 'Sample Class ID',
+          data: 'id',
+          type: 'read-only',
+          getDisplayValue: function(sampleClass) {
+            return sampleClass.id || 'Unsaved';
+          }
+        },
+        {
+          title: 'Alias',
+          data: 'alias',
+          type: 'text',
+          required: true,
+          maxLength: 255
+        },
+        {
+          title: 'Category',
+          data: 'sampleCategory',
+          type: 'dropdown',
+          required: true,
+          getSource: function() {
+            return Constants.sampleCategories;
+          },
```

Comment:
Is this not going to be bad for the DB if you change it?

Reply:
There's validation in the service layer that prevents changing it if the class is used by any samples, or if relationships would become invalid